actions/build-cxx-common/action.yml

```diff
@@ -0,0 +1,57 @@
+name: "Lifting tools CXX-Common"
+description: "Build cxx-common latest for lifting-tools"
+inputs:
+  target-export-path: # id of input
+    description: "where to export the cxx-common build"
+    required: true
+  gh-token:
+    description: "token for the target nuget cache"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+        path: cxx-common
+        repository: "lifting-bits/cxx-common"
+
+    - name: Initialize vcpkg
+      shell: bash
+      working-directory: cxx-common
+      run: |
+        { read -r vcpkg_repo_url && read -r vcpkg_commit; } <./vcpkg_info.txt || exit 1
+        git clone "${vcpkg_repo_url}"
+        git -C vcpkg checkout "${vcpkg_commit}"
+        echo "VCPKG_ROOT=$(pwd)/vcpkg" >> $GITHUB_ENV
+        echo "VCPKG_DISABLE_METRICS=1 >> $GITHUB_ENV"
+        echo "VCPKG_USE_NUGET_CACHE=1 >> $GITHUB_ENV"
+        echo "VCPKG_BINARY_SOURCES='clear;nuget,GitHub,readwrite;nugettimeout,3601' >> $GITHUB_ENV"
+    - name: "Bootstrap vcpkg"
+      shell: "bash"
+      working-directory: cxx-common
+      run: |
+        ./vcpkg/bootstrap-vcpkg.sh
+    - name: "Setup NuGet Credentials"
+      shell: "bash"
+      working-directory: cxx-common
+      run: |
+        mono `./vcpkg/vcpkg fetch nuget | tail -n 1` \
+          sources add \
+          -source "https://nuget.pkg.github.com/lifting-bits/index.json" \
+          -storepasswordincleartext \
+          -name "GitHub" \
+          -username "lifting-bits" \
+          -password "${{ inputs.gh-token }}"
+        mono `./vcpkg/vcpkg fetch nuget | tail -n 1` \
+          setapikey "${{ inputs.gh-token }}" \
+          -source "https://nuget.pkg.github.com/lifting-bits/index.json"
```

Comment:
Very weird it's still not working... In cxx-common, we `add` first and then `update`... I forget if that was for a special reason, like self-hosted runners, or if the authentication is finicky
https://github.com/lifting-bits/cxx-common/blob/94533d40904afb26ac979a8c91282f89f3c81fe3/.github/workflows/vcpkg_ci_amd64.yml#L98-L111
also, I run vcpkg from the same shell as the one I authenticate in 
https://github.com/lifting-bits/cxx-common/blob/94533d40904afb26ac979a8c91282f89f3c81fe3/.github/workflows/vcpkg_ci_amd64.yml#L117